File: FizzBuzz/Scala/src/main/scala/FizzBuzz.scala
object FizzBuzz {

  def stringFor(input: Any): Any = {
    val result = process(input)
    print(result)
  }

  private def process(input: Any): Any = {
    FilterJob.filter(input)
  }

  private def print(value: Any): Any = {
    PrinterJob.print(value)
  }
}

File: FizzBuzz/erlang/src/FizzBuzz.erl
-module('FizzBuzz').

-export([stringFor/1]).

stringFor(Input) ->
    Result = process(Input),
    print(Result).

process(Input) when is_list(Input) ->
    lists:map(fun filter/1, Input);
process(Input) when is_integer(Input) andalso Input > 0 ->
    filter(Input).

filter(Input) ->
    'FilterJob':filter(Input).

print(Value) ->
    Printers = 'PrintersProvider':printers(),
    'PrinterJob':print(Value, Printers).
